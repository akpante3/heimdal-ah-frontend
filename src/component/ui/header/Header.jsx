/* eslint-disable react/forbid-prop-types */
import React from 'react';
import './Header.scss';
import NavItems from '../navItems/NavItems';
import ActiveUser from '../activeuser/ActiveUser';
import Logo from '../logo/Logo';
import SearchForm from '../searchForm/SearchForm';
import HeaderButton from '../headerButton/HeaderButton';

/**
 * @param {boolean} isAuthenticated should be a boolean
 */
const Header = (props) => {
  const { isAuthenticated, user } = props;
  return (
    <header className="header">
      <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <Logo />
        <div className="collapse navbar-collapse" id="navbarCollapse">
          <ul className="navbar-nav mr-auto">
            <NavItems />
          </ul>
          {isAuthenticated ? <ActiveUser username={user.username} /> : <HeaderButton />}
          <SearchForm />
        </div>
      </nav>
    </header>
  );
};

export default Header;
