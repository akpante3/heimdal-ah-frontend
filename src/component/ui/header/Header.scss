$cardo: "Cardo", serif;
$dropdown-content: #f1f1f1;
$dropdown-content-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
$primary-color: #606e86;
$secondary-color: #ffff;
$reem-kufi: "Reem Kufi", sans-serif;

.header {
  font-family: $reem-kufi;

  .navbar {
    background: $secondary-color;
    box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.1);
  }

  .bg-light {
    background-color: $secondary-color !important;
  }

  .header-form {
    padding: 10px;
    margin-right: 20px;
    margin-top: -10px !important;
  }

  .header-input {
    display: none !important;
  }

  .active_user_logo {
    margin-right: 70px;
  }

  .notification_icon {
    margin-right: 60px;
    width: 35px;
  }

  .profile-icon {
    margin-left: 70px;
    margin-top: 20px;
    margin-bottom: 20px;
    width: 30px;
  }

  .search-input {
    margin-right: 10px !important;
    width: 150px;
  }

  .search_icon {
    -webkit-transform: scaleX(-1);
    color: $primary-color;
    cursor: pointer;
    transform: scaleX(-1);
  }

  .username {
    font-size: 18px;
    padding: 10px;

    &::after {
      border-bottom: 1px solid $primary-color;
      content: "";
      display: block;
      margin: 0 auto;
      margin-top: 12px;
      width: 90%;
    }
    text-align: center;
  }

  .dropdown {
    display: inline-block;
    position: relative;
  }

  .dropdown-content {
    background-color: $dropdown-content;
    box-shadow: $dropdown-content-shadow;
    display: none;
    font-family: "Cardo", serif;
    min-width: 160px;
    position: absolute;
    z-index: 1;

    a {
      color: $primary-color;
      display: block;
      padding: 12px 16px;
      text-decoration: none;
      &:hover {
        background-color: $primary-color;
        color: $secondary-color;
      }
    }
  }

  .user-settings::before {
    border-top: 1px solid $primary-color;
    content: "";
    display: block;
    margin: 0 auto;
    width: 90%;
  }

  .dropdown:hover .dropdown-content {
    display: block;
    margin-right: 60px;
  }
}
