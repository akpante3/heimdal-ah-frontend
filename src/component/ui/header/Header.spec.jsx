/* eslint-disable no-undef */
import React from 'react';
import Header from './Header.jsx';

describe('<Header />', () => {
  it('should render the Header component', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.length).toBe(1);
  });

  it('should test the children of the header', () => {
    const wrapper = shallow(<Header />);
    const children = wrapper.find('nav');
    expect(children.length).toBe(1);
  });
});
