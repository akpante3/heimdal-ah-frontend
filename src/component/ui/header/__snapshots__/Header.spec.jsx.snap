// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should test the children of the header 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <Logo />
        <div
          className="collapse navbar-collapse"
          id="navbarCollapse"
        >
          <ul
            className="navbar-nav mr-auto"
          >
            <NavItems />
          </ul>
          <HeaderButton />
          <SearchForm />
        </div>
      </nav>,
      "className": "header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <div
            className="collapse navbar-collapse"
            id="navbarCollapse"
          >
            <ul
              className="navbar-nav mr-auto"
            >
              <NavItems />
            </ul>
            <HeaderButton />
            <SearchForm />
          </div>,
        ],
        "className": "navbar navbar-expand-lg navbar-light bg-light",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ul
                className="navbar-nav mr-auto"
              >
                <NavItems />
              </ul>,
              <HeaderButton />,
              <SearchForm />,
            ],
            "className": "collapse navbar-collapse",
            "id": "navbarCollapse",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavItems />,
                "className": "navbar-nav mr-auto",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navbar navbar-expand-lg navbar-light bg-light"
        >
          <Logo />
          <div
            className="collapse navbar-collapse"
            id="navbarCollapse"
          >
            <ul
              className="navbar-nav mr-auto"
            >
              <NavItems />
            </ul>
            <HeaderButton />
            <SearchForm />
          </div>
        </nav>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Logo />,
            <div
              className="collapse navbar-collapse"
              id="navbarCollapse"
            >
              <ul
                className="navbar-nav mr-auto"
              >
                <NavItems />
              </ul>
              <HeaderButton />
              <SearchForm />
            </div>,
          ],
          "className": "navbar navbar-expand-lg navbar-light bg-light",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ul
                  className="navbar-nav mr-auto"
                >
                  <NavItems />
                </ul>,
                <HeaderButton />,
                <SearchForm />,
              ],
              "className": "collapse navbar-collapse",
              "id": "navbarCollapse",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NavItems />,
                  "className": "navbar-nav mr-auto",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "ul",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
